extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/tenant/HibernateMultiTenantConnectionProvider.java

```diff
@@ -33,12 +33,14 @@ protected ConnectionProvider getAnyConnectionProvider() {
     }
 
     @Override
-    protected ConnectionProvider selectConnectionProvider(String tenantIdentifier) {
+    protected ConnectionProvider selectConnectionProvider(final String tenantIdentifier) {
         LOG.debugv("selectConnectionProvider({0})", tenantIdentifier);
 
         ConnectionProvider provider = providerMap.get(tenantIdentifier);
         if (provider == null) {
-            return providerMap.computeIfAbsent(tenantIdentifier, tid -> resolveConnectionProvider(tid));
+            final ConnectionProvider connectionProvider = resolveConnectionProvider(tenantIdentifier);
```

Comment:
Are you sure you want to do this? It could introduce a race condition, because multiple parallel requests will use the same object. I think this is not thread safe.

Reply:
It seems fine to me - this is indeed a race condition but it's benign, as the end result doesn't change.

Reply:
Yes, this is correct. I just thought the "computeIfAbsent" is a more accurate usage of the API and does not cause such questions, because it is defined by the API of the hash map implementation.

Reply:
I agree with you - your solution was more elegant. But we try to avoid lambdas for efficiency reasons. Of course we're not extreme about it: lambdas are a fine solution when they are a good solution, but I still tend to prefer avoiding them in such situations.